raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -1396,12 +1398,12 @@ private FetchResponseData tryCompleteFetchRequest(
             int lastFetchedEpoch = request.lastFetchedEpoch();
             LeaderState<T> state = quorum.leaderStateOrThrow();
 
-            Optional<OffsetAndEpoch> latestSnapshotId = log.latestSnapshotId();
+            OffsetAndEpoch latestSnapshotId = log.latestSnapshotId().orElse(null);
             final ValidOffsetAndEpoch validOffsetAndEpoch;
-            if (fetchOffset == 0 && latestSnapshotId.isPresent()) {
-                // If the follower has an empty log and a snapshot exist, it is always more efficient
+            if (fetchOffset == 0 && latestSnapshotId != null && !latestSnapshotId.equals(BOOTSTRAP_SNAPSHOT_ID)) {
```

Comment:
Let's avoid the use of `null`.
```java
            if (fetchOffset == 0 &&
                latestSnapshotId.isPresent() &&
                !latestSnapshotId.get().equals(BOOTSTRAP_SNAPSHOT_ID)
            ) {
```

Reference: https://www.infoq.com/presentations/Null-References-The-Billion-Dollar-Mistake-Tony-Hoare/